### PR TITLE
User Query filters errors

### PR DESCRIPTION
**Pull request by TheBestPessimist**
https://github.com/metasfresh/metasfresh/issues/6476

### CODE REVIEW

File: de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/model/ColumnDisplayTypeProvider.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package org.compiere.model;
+
+public interface ColumnDisplayTypeProvider
```
Comment (metas-ts) [inline de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/model/ColumnDisplayTypeProvider.java:25]: i suggest you flag this as `@FunctionalInterface`, so we can inject lambdas

File: de.metas.adempiere.adempiere/base/src/main/java/org/compiere/apps/search/UserQueryFieldHelper.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+			{
+				return DisplayType.toBooleanString((Boolean)valueObj);
+			}
+		}
+		catch (final Exception ex)
+		{
+			String error = ex.getLocalizedMessage();
+			if (error == null || error.isEmpty())
+			{
+				error = ex.toString();
```
Comment (metas-ts) [inline de.metas.adempiere.adempiere/base/src/main/java/org/compiere/apps/search/UserQueryFieldHelper.java:96]: i think here you lost the part from the legacy code where the error is logger. pls add some logging; (Level.WARN)

Comment (TheBestPessimist) [reply de.metas.adempiere.adempiere/base/src/main/java/org/compiere/apps/search/UserQueryFieldHelper.java:96]: Added error logs

File: de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/model/ColumnDisplayTypeProvider.java (hunk 1 above)
Comment (TheBestPessimist) [reply de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/model/ColumnDisplayTypeProvider.java:25]: done.

**Pull request merged**